### ISSUE TITLE
space-before-function-paren is not working pretty good with async es6 function

### ISSUE THREAD

**Issue by muuvmuuv** [babel/babel-eslint#751]
It looks like eslint has problems with async function that uses arrow functions. It throws a error where IMO it should not or am I missing an eslint option?

Error: `Unexpected space before function parentheses.eslint(space-before-function-paren)`
Config: https://hastebin.com/oliyavicec.json
Code:
```js
test('return a translated text string for prop TEST into german (de-DE)', async () => {
  __.language = 'de-DE';                                                  ^^^^^
  const translatedString = await __.text('TEST');

  expect(translatedString).toBe('Tests');
});
```

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [`@babel/eslint-parser` has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
